utils_for_nautilus.py

```diff
@@ -0,0 +1,190 @@
+from abc import ABC, abstractmethod
+from Bio.SeqUtils import gc_fraction
+from Bio.Seq import Seq
+from Bio import SeqIO
+
+class BiologicalSequence(ABC):
+    """
+    Abstract base class for biological sequences.
+
+    Provides a common interface for different types of biological sequences.
+    """
+    @abstractmethod
+    def __len__(self):
+        pass
+
+    @abstractmethod
+    def __repr__(self):
+        pass
+
+    @abstractmethod
+    def check_alphabet(self):
+        pass
+
+    @abstractmethod
+    def __getitem__(self, index):
+        pass
+
+    @abstractmethod
+    def __str__(self):
+        pass
+
+class NucleicAcidSequence(BiologicalSequence):
+    """
+    Represents a nucleic acid sequence.
+
+    Provides methods for working with nucleic acid sequences, including calculating the complement and reverse complement.
+    """
+    def __init__(self,seq,dict):
+        self.seq = seq
+        self.dict = dict
+    
+    def __len__(self):
+        return len(self.seq)
+    
+    def __repr__(self):
+        return f'your sequence: {self}' 
+
+    def check_alphabet(self):
+        valid_bases = set("ATGCUatgcu")
+        return all(base in valid_bases for base in self.seq)
+
+    def __getitem__(self, index):
+        return self.seq[index]
+    
+    def __str__(self):
+        return self.seq
+
+    def complement(self):
+        result = ""
+        for letter in self.seq:
+            result += self.dict[letter] 
+        return result
+    
+    def reverse(self):
+        return self.seq[::-1]
+    
+    def reverse_complement(self):
+        complemented = self.complement()
+        return complemented[::-1]
+    
+class DNASequence(NucleicAcidSequence):
+    """
+    Represents a DNA sequence.
+
+    Provides methods specific to DNA sequences, including transcription.
+    """
+    def __init__(self,seq):
+        dna_complement_dict = {
+        "A": "T",
+        "T": "A",
+        "G": "C",
+        "C": "G",
+        "a": "t",
+        "t": "a",
+        "g": "c",
+        "c": "g",
+    }
+        super().__init__(seq,dna_complement_dict)
+        
+    def transcribe(self):
+        return self.seq.replace('T', 'U').replace('t', 'u')
+    
+class RNASequence(NucleicAcidSequence):
+    """
+    Represents an RNA sequence.
+
+    Provides methods specific to RNA sequences.
+    """
+    def __init__(self,seq):
+        rna_complement_dict = {
+        "A": "U",
+        "U": "A",
+        "G": "C",
+        "C": "G",
+        "a": "u",
+        "u": "a",
+        "g": "c",
+        "c": "g",
+    }
+        super().__init__(seq,rna_complement_dict)
```

Comment:
```suggestion
        super().__init__(seq, rna_complement_dict)
```
здесь и выше в подобных случаях не хватает пробелов

-1